crates/wit-component/src/printing.rs

```diff
@@ -50,35 +50,41 @@ impl WitPrinter {
         self
     }
 
-    /// Print the given WIT package to a string.
-    pub fn print(&mut self, resolve: &Resolve, pkgid: PackageId) -> Result<String> {
-        let pkg = &resolve.packages[pkgid];
-        self.print_docs(&pkg.docs);
-        self.output.push_str("package ");
-        self.print_name(&pkg.name.namespace);
-        self.output.push_str(":");
-        self.print_name(&pkg.name.name);
-        if let Some(version) = &pkg.name.version {
-            self.output.push_str(&format!("@{version}"));
-        }
-        self.print_semicolon();
-        self.output.push_str("\n\n");
-        for (name, id) in pkg.interfaces.iter() {
-            self.print_docs(&resolve.interfaces[*id].docs);
-            self.output.push_str("interface ");
-            self.print_name(name);
-            self.output.push_str(" {\n");
-            self.print_interface(resolve, *id)?;
-            writeln!(&mut self.output, "}}\n")?;
-        }
+    /// Print a set of one or more WIT packages into a string.
+    pub fn print(&mut self, resolve: &Resolve, pkg_ids: Vec<PackageId>) -> Result<String> {
```

Comment:
Conventionally a Rust array is typically passed as `&[T]` instead of `Vec<T>` to avoid taking unnecessary ownership.

Additionally should this method dynamically switch between the `package foo:bar;` syntax and `package foo:bar { ... }` depending on how many entries there are in `pkg_ids`?

Reply:
> Conventionally a Rust array is typically passed as `&[T]` instead of `Vec<T>` to avoid taking unnecessary ownership.

Whoops, thanks for catching that! Fixed.

> Additionally should this method dynamically switch between the `package foo:bar;` syntax and `package foo:bar { ... }` depending on how many entries there are in `pkg_ids`?

I've made this change, though in doing so I noticed that we have no "pretty printing" tests in and of themselves? The printer seems to only be used in the service of other tools. As currently constituted, there are no tools (in this repo anyway) that would be able to emit multi-package output. For example, `component embed` requires a `--world` argument for multi-package input, narrowing the result to a single pacakge.

We could of course make pretty printing tests to cover this sort of thing more directly. Speaking of - does WIT currently have an official formatter? I couldn't find one, but that doesn't mean one does not exist.

Reply:
For the testing bits, actually if you're up for it I think it'd be great to add a test. I think it'd work well in `tests/cli/*.wit` and the `// RUN: ...` annotation at the top would basically say "please re-print this WIT file". That way you could feed in a multi-package input and the "golden output" would be a multi-package one as well. You're right we're not testing this super well though!

For formatting, not at this time. I know some folks were working on a vscode extension but I'm not sure if it did formatting.

Reply:
Can I do this in a follow up PR? This one has already gotten a bit unwieldy, and I'd like to get it merged rather than repeatedly rebasing. It seems like adding test coverage is a separate concern anyway.

Reply:
Sounds good to me 👍